drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1118 @@
+		return -EINVAL;
+	}
+
+	CHECKIF(val == NULL) {
+		LOG_ERR("val: NULL");
+		return -EINVAL;
+	}
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	int retval;
+
+	switch ((uint16_t)attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		if (drv_cfg->meas_range != TMAG5273_DT_AXIS_RANGE_RUNTIME) {
+			return -ENOTSUP;
+		}
+
+		retval = tmag5273_attr_set_xyz_meas_range(dev, val);
+		if (retval < 0) {
+			return retval;
+		}
+		break;
+	case TMAG5273_ATTR_ANGLE_MAG_AXIS:
+		if (drv_cfg->angle_magnitude_axis != TMAG5273_DT_ANGLE_MAG_RUNTIME) {
+			return -ENOTSUP;
+		}
+
+		retval = tmag5273_attr_set_xyz_calc(dev, val);
+		if (retval < 0) {
+			return retval;
+		}
+		break;
+	default:
+		LOG_ERR("unknown attribute %d", attr);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int tmag5273_attr_get(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, struct sensor_value *val)
+{
+	CHECKIF(dev == NULL) {
+		LOG_ERR("dev: NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(val == NULL) {
+		LOG_ERR("val: NULL");
+		return -EINVAL;
+	}
+
+	if (chan != SENSOR_CHAN_MAGN_XYZ) {
+		return -ENOTSUP;
+	}
+
+	const struct tmag5273_config *drv_cfg = dev->config;
+
+	int retval;
+
+	switch ((uint16_t)attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		if (drv_cfg->meas_range != TMAG5273_DT_AXIS_RANGE_RUNTIME) {
+			return -ENOTSUP;
+		}
+
+		retval = tmag5273_attr_get_xyz_meas_range(dev, val);
+		if (retval < 0) {
+			return retval;
+		}
+		break;
+	case TMAG5273_ATTR_ANGLE_MAG_AXIS:
+		if (drv_cfg->angle_magnitude_axis != TMAG5273_DT_ANGLE_MAG_RUNTIME) {
+			return -ENOTSUP;
+		}
+
+		retval = tmag5273_attr_get_xyz_calc(dev, val);
+		if (retval < 0) {
+			return retval;
+		}
+		break;
+	default:
+		LOG_ERR("unknown attribute %d", attr);
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int tmag5273_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	int retval;
+
+	uint8_t i2c_buffer[TMAG5273_REG_RESULT_END - TMAG5273_REG_RESULT_BEGIN + 2] = {0};
+
+	/* trigger a conversion and wait until done if in standby mode */
+	if (drv_cfg->operation_mode == TMAG5273_DT_OPER_MODE_STANDBY) {
+		if (drv_cfg->trigger_conv_via_int) {
+			retval = tmag5273_dev_int_trigger(drv_cfg);
+			if (retval < 0) {
+				return retval;
+			}
+		}
+
+		uint8_t conv_bit = TMAG5273_CONVERSION_START_BIT;
+
+		i2c_buffer[0] = 0;
+
+		while ((i2c_buffer[0] & TMAG5273_RESULT_STATUS_MSK) !=
+		       TMAG5273_CONVERSION_COMPLETE) {
+			retval = i2c_reg_read_byte_dt(
+				&drv_cfg->i2c, TMAG5273_REG_CONV_STATUS | conv_bit, &i2c_buffer[0]);
+
+			if (retval < 0) {
+				LOG_ERR("error reading conversion state %d", retval);
+				return retval;
+			}
+
+			conv_bit = 0;
+
+			k_usleep(drv_data->conversion_time_us);
+		}
+	}
+
+	/* read data */
+	uint8_t start_address = 0;
+	uint8_t end_address = 0;
```

Comment:
no need to initialize these, you are just silencing a potentially useful compiler warning